### ISSUE TITLE
What is the behavior of ReloadableProgram?

### ISSUE THREAD

**Issue by mahroni-14** [moderngl/moderngl-window#84]
I have created a **Custom** Window Context with **GLFW**, where I have needed to create a **single file** shader program with below code:

**`resources.programs.load(ProgramDescription(path = shader_path, reloadable = True))`**

I am unable to understand the behavior of the code. Does it suppose to reload the shader program **auto magically** or I am missing something?

**Comment by einarf** [comment]
That one is lingering from the my original project this one is based on. It was a bit more frameworky so there was a default `R` key mapped in debug mode that would reload all programs marked as reloadable. This will only work if you are rendering with `VAO` instances.

I had completely forgotten about this and it might be possible to salvage it. It's a very useful feature to be able to reload programs while running.

**Comment by mahroni-14** [comment]
Thank you, for your quick answer.

So, for the default **R key** behavior do I need to use the **Default camera**?
If so, how can I use this behavior from my code. Is there any **utility method**?
You see I am trying to make a **ShaderToy** like system but whenever a glsl file change,
I need it to be re-create the program.

**Comment by einarf** [comment]
The reload stuff is not really working well, so I would abandone using the `reloadable=True` system for now. We should keep this one open as a reminder to fix this issue.

In the meantime you can look at : https://github.com/einarf/shadertoy

**Comment by mahroni-14** [comment]
What you have made it already?? :D

**Comment by einarf** [comment]
Of course. It's not perfect, but it works 😄

**Comment by mahroni-14** [comment]
Do you use any kind of **file watcher**?

**Comment by einarf** [comment]
Nope. It just stats the files in intervals. A bit lazy, but I needed it to work cross platform. There was some bug with a library I was using for it and I just took the simple path for now.

**Comment by mahroni-14** [comment]
Thank you.

So, do you have any future plan about **ReloadableProgram** support with **watchdog** or something like that for **moderngl-window**?

**Comment by einarf** [comment]
It's an immensely useful feature, so the plan is definitely to add it when there is time. This issue can be a reminder that it needs to be done.
